Collections model binding "index" parameter warning (#24175)
dotnet/AspNetCore.Docs: aspnetcore/mvc/models/model-binding.md

  [a]=1050&[b]=2000&index=a&index=b
  ```

>  Avoid binding a parameter or a property named `index` or `Index` if it is adjacent to a collection value. Model binding attempts to use `index` as the index for the collection which might result in incorrect binding. For example, consider the following action:
>   
>   ```csharp
>   public IActionResult Post(string index, List<Product> products)

Reviewer: ```suggestion
  Avoid binding a parameter or a property named `index` or `Index` if it is adjacent to a collection value. Model binding attempts to use `index` as the index for the collection which might result in incorrect binding. For example, consider the following action:
  
  ```csharp
  public IActionResult Post(string index, List<Product> products)
  ```
  In the preceding code, the `index` query string parameter binds to the `index` method parameter and also is used to bind the product collection. Renaming the `index` parameter or using a model binding attribute to configure binding avoids this issue:
  ```csharp
  public IActionResult Post(string productIndex, List<Product> products)
  ```
```

  [a]=1050&[b]=2000&index=a&index=b
  ```

> <!-- STUB: Make same change here -->

Reviewer: ```suggestion
 Avoid binding a parameter or a property named `index` or `Index` if it is adjacent to a collection value. Model binding attempts to use `index` as the index for the collection which might result in incorrect binding. For example, consider the following action:
  
  ```csharp
  public IActionResult Post(string index, List<Product> products)
  ```
  In the preceding code, the `index` query string parameter binds to the `index` method parameter and also is used to bind the product collection. Renaming the `index` parameter or using a model binding attribute to configure binding avoids this issue:
  ```csharp
  public IActionResult Post(string productIndex, List<Product> products)
  ```
```

  ```
  [a]=1050&[b]=2000&index=a&index=b
  ```

>   > [!WARNING] 

Reviewer 2: Maybe we could remove the warning and simply state it? It's not really a error case.

Author: @pranavkm, we could if you prefer but I think that effect of using the index parameter cause enough disturbance, since the list will be populated with an item that could potentially break the user's code.

Reviewer: Can you check [my commit](https://github.com/dotnet/AspNetCore.Docs/pull/24175/commits/86fed85d8b11e542770b09466ae74d0d681f970c)?

Author: I like that, it is pretty clear. Thanks both for the help

>   > Since the model binding looks for the keyword `index` from a Form or Query string is strongly recommended to not use it as your action's parameter name to avoid unexpected behaviors.

Reviewer 2: ```suggestion
  > Avoid binding a parameter or a property named `index` or `Index` if it is adjacent to a collection value. Model binding will attempt to use it as the index for the collection which might result in incorrect binding. For example, consider the following action:
  
  ```csharp
  public IActionResult Post(string index, List<Product> products)
 ```
The `index` parameter will bound to the parameter and will also be used to bind the `product` collection which might not be what is desired. Renaming the parameter or using a model binding attribute to configure the binding would avoid this issue.
